### PR TITLE
Added wavelet transform to FloatVectorList.

### PR DESCRIPTION
**Pull request by niklasborwell**

### CODE REVIEW

File: source/math/FloatVectorList.ooc [hunk 1]
```diff
@@ -198,4 +198,23 @@ FloatVectorList: class extends VectorList<Float> {
 		forwardGaussian free()
 		result
 	}
+	getWaveletTransform: func(levels: Int) -> VectorList<This> {
```
Comment (davidhesselbom) [inline source/math/FloatVectorList.ooc]: `func (`

**Comment by niklasborwell** [comment]
Updated.

File: source/math/FloatVectorList.ooc [hunk 2]
```diff
@@ -198,4 +198,23 @@ FloatVectorList: class extends VectorList<Float> {
...
+		result := VectorList<This> new(levels)
+		previous := this
+		filtered: This
+		for (level in 0..levels) {
+			size := 1 + pow(2, level + 1)
+			if (filtered != null)
+				filtered free()
+			kernel := This gaussianKernel(size)
+			filtered := this convolve(kernel)
+			kernel free()
```
Comment (davidhesselbom) [inline source/math/FloatVectorList.ooc]: Is `kernel` at all affected by the `this convolve(kernel)` call? If not, you may as well keep it between the loop iterations.

Comment (davidhesselbom) [inline source/math/FloatVectorList.ooc]: Oh wait, it depends on `size`, which isn't constant. Nevermind...

File: source/math/FloatVectorList.ooc [hunk 3]
```diff
@@ -198,4 +198,23 @@ FloatVectorList: class extends VectorList<Float> {
...
+	getWaveletTransform: func (levels: Int) -> VectorList<This> {
+		result := VectorList<This> new(levels)
+		previous := this
+		filtered: This
+		for (level in 0..levels) {
+			size := 1 + pow(2, level + 1)
+			if (filtered != null)
+				filtered free()
+			kernel := This gaussianKernel(size)
+			filtered := this convolve(kernel)
```
Comment (davidhesselbom) [inline source/math/FloatVectorList.ooc]: why `:=` instead of `=`?

File: source/math/FloatVectorList.ooc [hunk 4]
```diff
@@ -198,4 +198,23 @@ FloatVectorList: class extends VectorList<Float> {
 		forwardGaussian free()
 		result
 	}
+	getWaveletTransform: func (levels: Int) -> VectorList<This> {
+		result := VectorList<This> new(levels)
+		previous := this
+		filtered: This
```
Comment (davidhesselbom) [inline source/math/FloatVectorList.ooc]: `filtered: This = null`, just to be safe.

Comment (niklasborwell) [inline source/math/FloatVectorList.ooc]: wait, what? Whats the purpose of that?

Comment (davidhesselbom) [inline source/math/FloatVectorList.ooc]: OK, this is the C output:
```
    #line 202 "/home/dhesselbom/versioned/ooc/ooc-kean/source/math/FloatVectorList.ooc"
    VectorList__VectorList* result = VectorList__VectorList_new_heap((lang_types__Class*)FloatVectorList__FloatVectorList_class(), levels);
    #line 203 "/home/dhesselbom/versioned/ooc/ooc-kean/source/math/FloatVectorList.ooc"
    FloatVectorList__FloatVectorList* previous = this;
    #line 204 "/home/dhesselbom/versioned/ooc/ooc-kean/source/math/FloatVectorList.ooc"
    FloatVectorList__FloatVectorList* filtered = NULL;
```
So I guess ooc can handle it after all. Still, it's good practice to always initialize your variables before using them.

**Pull request merged**